packages/adblocker/src/filters/network.ts

```diff
@@ -400,18 +397,169 @@ function compileRegex(
   return new RegExp(filter);
 }
 
-export function findLastIndexOfUnescapedCharacter(text: string, character: string) {
-  let lastIndex = text.lastIndexOf(character);
+/**
+ * Collects a filter option key until the function sees the special character.
+ * This function will stop iterating over the given string if it sees equal sign or comma sign.
+ * If there's an equal sign, it means that we'll see the value.
+ * Otherwise, if there's a comma sign, it means that the option doesn't have any values.
+ * Note that this function doesn't respect the escaping sign.
+ */
+function getFilterOptionKey(line: string, pos: number, end: number) {
+  let code: number;
+  let value = '';
+
+  for (; pos < end; pos++) {
+    code = line.charCodeAt(pos);
+
+    if (code === 61 /* '=' */ || code === 44 /* ',' */) {
+      break;
+    }
+
+    value += line.charAt(pos);
```

Comment:
Perhaps safer to avoid string concatenation in a loop (if a JavaScript engine implements it as O(n^2)).

Search only for the separator (`=` or `,`):
* it was found, you can use slice from pos to this position
* otherwise, from pos to end

Reply:
@philipp-classen I refactored functions not to rely on string concatenation in [`bf6eada` (#3897)](https://github.com/ghostery/adblocker/pull/3897/commits/bf6eada804037f3582deb2ca49d56a94f57b307d). Please, take a review on the new changes.